gtest/src/wasm_executor.rs

```diff
@@ -40,58 +58,68 @@ impl WasmExecutor {
         };
 
         let config = Config::new();
-        let engine = Engine::new(&config).expect("Failed to create engine");
+        let engine = Engine::new(&config)?;
         let mut store = Store::<StoreData<Ext>>::new(&engine, store_data);
 
-        let module = Module::new(&engine, meta_binary).expect("Failed to create module");
+        let module = Module::new(&engine, meta_binary)?;
 
         let mut memory =
-            WasmtimeMemory::new(&mut store, MemoryType::new(program.static_pages().0, None))
-                .expect("Failed to create memory");
+            WasmtimeMemory::new(&mut store, MemoryType::new(program.static_pages().0, None))?;
+
         let funcs = funcs_tree::build(&mut store, memory, None);
         let mut externs = Vec::with_capacity(module.imports().len());
```

Comment:
BTW, all of this logic can be replaced with `wasmtime::Linker`

Reply:
Not very familiar with `wasmtime`. Please check this commit to whether I understand you correctly:

https://github.com/gear-tech/gear/pull/1106/commits/8f1374d39e364ca8eb5533a338d97b4536b78ff3

Reply:
Oh, I'm sorry. I thought `Linker` can do something more. You used it correctly